config/config.go

```diff
@@ -227,3 +301,64 @@ func highlightError(data []byte, pos int64) (int, int, string) {
 	}
 	return line, int(col), fmt.Sprintf("%s%s%s", prevLine, thisLine, highlight)
 }
+
+func decodeArray(raw interface{}) []interface{} {
+	if raw == nil {
+		return nil
+	}
+	var arr []interface{}
+	switch reflect.TypeOf(raw).Kind() {
+	case reflect.Slice:
+		s := reflect.ValueOf(raw)
+		for i := 0; i < s.Len(); i++ {
+			v := s.Index(i)
+			if !v.IsNil() {
+				arr = append(arr, v.Interface())
+			}
+		}
+		return arr
+	}
+	return nil
+}
+
+// We can't use mapstructure to decode our config map since we want the values
+// to also be raw interface{} types. mapstructure can only decode
```

Comment:
Oh, I see. I was missing where we'd of course be passing in an actual implementation of `interface{}` to `DecodeRaw`.  Makes sense.